sql/bench_test.go

```diff
@@ -323,6 +319,89 @@ func BenchmarkUpdate100_Postgres(b *testing.B) {
 	benchmarkPostgres(b, runBenchmarkUpdate100)
 }
 
+// runBenchmarkDelete benchmarks deleting count rows from a table.
+func runBenchmarkDelete(b *testing.B, db *sql.DB, rows int) {
+	if _, err := db.Exec(`DROP TABLE IF EXISTS bench.delete`); err != nil {
+		b.Fatal(err)
+	}
+	if _, err := db.Exec(`CREATE TABLE bench.delete (k INT PRIMARY KEY)`); err != nil {
+		b.Fatal(err)
+	}
+	defer func() {
+		if _, err := db.Exec(`DROP TABLE bench.delete`); err != nil {
+			b.Fatal(err)
+		}
+	}()
+
+	b.ResetTimer()
+	var buf bytes.Buffer
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		buf.Reset()
+		buf.WriteString(`INSERT INTO bench.delete VALUES `)
+		for j := 0; j < rows; j++ {
+			if j > 0 {
+				buf.WriteString(", ")
+			}
+			fmt.Fprintf(&buf, "(%d)", j)
+		}
+		if _, err := db.Exec(buf.String()); err != nil {
+			b.Fatal(err)
+		}
+		b.StartTimer()
```

Comment:
should this happen just before the db.Exec? slightly inconsistent with other impls here, but still seems preferable.